src/guiguts/tools/jeebies.py

```diff
@@ -0,0 +1,297 @@
+
+logger = logging.getLogger(__package__)
+
+DEFAULT_DICTIONARY_DIR = importlib.resources.files(dictionaries)
+PARANOID_LEVEL_3WORDS = 1.0
+
+_the_jeebies_checker = None  # pylint: disable=invalid-name
+
+
+class DictionaryNotFoundError(Exception):
+    """Raised when no hebe-forms dictionary found."""
+
+    def __init__(self, file: str) -> None:
+        self.file = file
+
+
+class JeebiesChecker:
+    """Provides jeebies check functionality."""
+
+    def __init__(self) -> None:
+        """Initialize SpellChecker class."""
+        self.dictionary: dict[str, int] = {}
+        self.load_phrases_file_into_dictionary()
+
+    def check_for_jeebies_in_file(self) -> None:
+        """Check for jeebies in the currently loaded file."""
+
+        # Create the checker dialog to show results
+        checker_dialog = CheckerDialog.show_dialog(
+            "Jeebies Results", rerun_command=jeebies_check
+        )
+        checker_dialog.reset()
+
+        # Get the whole of the file from the main text widget
+        text = maintext().get_text()
+        input_lines = text.splitlines()
+        # Ensure last paragraph converts to a line of text
+        input_lines.append("")
+
+        # Paragraphs as single string, case preserved
+        wbs = []
+        # Paragraphs as single string, all lower case
+        wbl = []
+
+        # for line, line_num in maintext().get_lines():
+        s = ""
+        be_cnt_in_file = 0
+        he_cnt_in_file = 0
+        for line in input_lines:
+            line = line.rstrip("\r\n")
+            be_cnt_in_file += len(re.findall(r"(?<=^|\W)be(?=\W|$)", line))
+            he_cnt_in_file += len(re.findall(r"(?<=^|\W)he(?=\W|$)", line))
+            if line == "":
+                # Blank line so end of paragraph
+                if s != "":
+                    # Something in the paragraph string so save it
+                    # after trimming leading and trailing space.
+                    s = s.strip()
+                    wbs.append(s)
+                    wbl.append(s.lower())
+                    s = ""
+            else:
+                # Still in paragraph
+                s += " " + line
+
+        if (be_cnt_in_file + he_cnt_in_file) == 0:
+            checker_dialog.add_entry(
+                f"  --> 'be' counted {be_cnt_in_file} times and 'he' counted {he_cnt_in_file} times in file."
+            )
+        else:
+            checker_dialog.add_entry(
+                f"  --> 'be' counted {be_cnt_in_file} times and 'he' counted {he_cnt_in_file} times in file. Looking for suspect phrases..."
+            )
+        checker_dialog.add_entry("")
+
+        be_suspects_cnt = self.find_and_report_be_phrases(wbs, wbl, checker_dialog)
+        he_suspects_cnt = self.find_and_report_he_phrases(wbs, wbl, checker_dialog)
+
+        if be_suspects_cnt == 0 and he_suspects_cnt == 0:
+            checker_dialog.add_entry("    No suspect phrases found.")
+
+    def find_and_report_be_phrases(
```

Comment:
Since this and `find_and_report_he_phrases()` are identical except for the `he` and `be` strings consider merging them into one so we don't need to worry about them drifting. You could even make this the core function `_find_and_report_xx_phrases(self, phrase: string, ...)` and have named wrapper functions that call them with `he` or `be`.

Reply:
@G4OEU - I noticed that your new version still has two very similar copies of the code, although they are now both contained within the same function.

I interpreted @cpeel's suggestion as meaning combine them into one version of the code, but using variables in place of "he" and "be", so `hebe` for "he" or "be" as you have used, and `behe` for "be" or "he" respectively. 
So it could look something like this (untested - just grabbed and edited from your version - I may easily have hebe and behe the wrong way round in places):

```python
    def find_and_report_hebe_phrases(
        self, hebe: str, wbs: list, wbl: list, checker_dialog: CheckerDialog
    ) -> int:
        """Look for suspect "w1 be w2" or "w1 he w2" phrases in paragraphs.
        """

        suspects_count = 0
        # Get the opposite word to the one being reported on
        behe = "he" if hebe == "be" else "be"

        # Search for 3-form pattern "w1 he/be w2" in a lower-case copy of paragraph.
        for rec_num, para_lc in enumerate(wbl):
            for match_obj in re.finditer(fr"[a-z’]+ {hebe} [a-z’]+", para_lc):
                # Have a 3-word form here ("must be taken").
                # See if it's in the list.
                hebe_form = para_lc[match_obj.start(0) : match_obj.end(0)]
                hebe_count = self.find_in_dictionary(be_form)
                # Change 'be' to 'he' and lookup that 3-word form.
                behe_form = re.sub(hebe, behe, be_form)
                behe_count = self.find_in_dictionary(he_form)
                # At this point we have the two forms and how common they are in
                # hebe_count and behe_count.

                # Improved Golang/PPWB format and values calculation
                if behe_count > 0 and (
                    hebe_count == 0 or behe_count / hebe_count > PARANOID_LEVEL_3WORDS
                ):
                    suspects_count += 1
                    info = f"({behe_count}/{hebe_count} - '{behe}' is seen more frequently than '{hebe}' in this phrase)"

                    # Add this 3-form and its info to the dialog

                    header = f'Query phrase "{hebe_form}" {info}'
                    # Get whole-words slice from original para text that is approximately
                    # centered on the 3-form phrase.
                    _, _, centered_slice = self.get_para_slice(
                       match_obj.start(0), wbs[rec_num]
                    )
                    self.add_to_dialog(header, centered_slice, checker_dialog)
        return suspects_count
```